chromium_src/components/search_engines/template_url_prepopulate_data.cc

```diff
@@ -186,6 +368,23 @@ GetBravePrepopulatedEnginesForCountryID(
 
 }  // namespace
 
+// Some engines (like Yahoo) have different URLs per country
+// The intention of this function is to find the generic one
+// (ex: PREPOPULATED_ENGINE_ID_YAHOO) and then substitute the
+// country specific version.
+void LocalizeEngineList(
+    std::vector<BravePrepopulatedEngineID>* engines,
+    int country_id) {
+  for (size_t i = 0; i < engines->size(); ++i) {
+    if ((*engines)[i] == PREPOPULATED_ENGINE_ID_YAHOO) {
+      const auto& it = yahoo_engines_by_country_id_map.find(country_id);
```

Comment:
this can be done just once outside the loop. So if there is nothing for given `country_id` no need to run it

Reply:
The intention is to be able to localize any engine for any country. The list may not even have Yahoo. For a more complex example, we could retrofit this in for DDG since there are different URLs used between US/DE/NZ, AU, IE... and in that case, the method would need to check list 1) for two different options and 2) see if there is a replacement for each